### DISCUSSION TITLE
Create 3 hardware?

### DISCUSSION THREAD

**Discussion by dtwind62** [iRobotEducation/create3_docs#313] [Questions]
Does the create 3 have some kind of microcontroller built into it? My understanding is the robot will  do simple tasks like dock/undock without a raspberry pi installed or is the raspberry pi the only processor/controller in it as it comes from the factory. It must at least have some kind of motor controller for the wheels to connect to.

**Comment by shamlian** [comment]
The Create 3 robot is built on the iRobot Roomba i3 platform; it contains a main microprocessor for ROS 2 and iRobot Coding basic command and control (and other high level sensor fusion and to control actions) and some microcontrollers for things like motor control. None of these processors are designed for the user to run arbitrary code upon. The robot can be teleoperated using [iRobot Coding](code.irobot.com), the [Python Web Playground](python.irobot.com), or any machine running ROS 2 without adding any extra hardware to the robot. The robot does not come with any third party development boards (like [Raspberry Pi](https://iroboteducation.github.io/create3_docs/hw/rpi_hookup/), [NVIDIA Jetson](https://iroboteducation.github.io/create3_docs/hw/jetson_hookup/), or others), but, depending on current draw, it is often possible to interface them directly to the robot if additional on-board computation is desired using the included [adapter board](https://iroboteducation.github.io/create3_docs/hw/adapter/).

**Comment by dtwind62** [comment]
Thank you for the information. I know it's not user programmable, I'm
learning about electronics and I was just wondering, I heard it's in the
Arm Cortex family of processors.
